docs/tutorials/craft-a-snap.rst

```diff
@@ -0,0 +1,549 @@
+.. _tutorial-craft-a-snap:
+
+Craft a snap
+============
+
+In this tutorial, we'll build a snap package for a Python app called pyfiglet. The
+concepts we'll cover transfer to both simple and complex snaps. We'll cover everything
+from creating the build environment and the configuration file, to troubleshooting
+missing libraries and finding which interfaces to open.
+
+It should take 20 minutes to complete.
+
+You won't need to come prepared with deep knowledge of software packaging, but
+familiarity with Linux paradigms and terminal operations is required.
+
+Once you complete this tutorial, you'll have experience hand-crafting snaps that will be
+the basis for further work with creating snaps.
+
+
+Lesson plan
+-----------
+
+Put simply, this tutorial is a run through of the process of constructing a snap. We
+show you how to:
+
+- Start a snap project from scratch
+- State the project's essential information
+- Define the project source files and main program
+- Package the snap
+- Modify the build process to enhance the snap's contents
+- Share system resources with the snap
+
+
+What we'll work with
+--------------------
+
+The object of this tutorial is to package `pyfiglet
+<https://github.com/pwaller/pyfiglet>`_ as a personal test snap. It's a lightweight app
+for displaying text as ASCII art, and is simple to build and test.
+
+
+What you'll need
+----------------
+
+For this tutorial, you'll need:
+
+- A x64 system running Ubuntu 22.04 or Ubuntu 24.04
+- A local user with super user privileges
+- 20GB of free storage
+
+
+Install Snapcraft and LXD
```

Comment:
It would be really nice to link out to the LXD documentation and mention Multipass as an alternative.  

Eventually we should add a link to https://github.com/canonical/craft-providers/issues/583.

I'm sensitive to this part of the tutorial because it's where most bad first impressions occur.  There's a fair chance LXD doesn't work out-of-the-box and it's common for users to try snapcraft and give up after getting an error within 60 seconds of using the tool.

Reply:
As part of my diligence, let's see if we can provide a workaround. :) I'd rather not partition the flow to Multipass if we can avoid it.

Is there something about the LXD installation process that's janky that we could smooth over? It was seamless on Ubuntu 24.04. Is it worse on other distros?

Reply:
Let me know what you think: 50bb010d9c59071fe120f6b8924d569925f4299b